### DISCUSSION TITLE
Wouldn't it be better to tie 'Game Mode' to the particular character instead of handling it globally?

### DISCUSSION THREAD

**Discussion by julealgon** [diasurgical/DevilutionX#7363] [Ideas]
I saw this change from @kphoenix137 and this came to my mind:
- https://github.com/diasurgical/devilutionX/pull/7362

It made me wonder... was there a strong reason not to tie the game mode to the character? For example, have a couple checkboxes on character creation (similar to how the "hardcode" option works in D2) instead of having these be global settings.

Doesn't this being global open the door for lots of weird issues when a character is created in one mode and "loaded" in another?

I'd imagine if the setting was tied to the character (and could not be changed) it would lead to a more robust experience (although less flexible).

Maybe even if it _was_ possible to change on a given character, at least it would give an opportunity to warn the player of some things that could change, instead of it just "happening" without warning.

It just feels very unintuitive to me to play in Hellfire mode, grab a bunch of Hellfire-exclusive items, and then load that same character in Diablo mode. IMHO that should not be allowed at all by default.

Thoughts?

**Comment by kphoenix137** [comment]
100% agree

**Comment by AJenbo** [comment]
> Doesn't this being global open the door for lots of weird issues when a character is created in one mode and "loaded" in another?

This is supported

**Comment by kphoenix137** [comment]
I very much like the idea of game mode being set on character creation, with no settings game mode option being required. Although that does mean the hellfire menu logo and video wouldn't get displayed, some hellfire users may dislike that

**Reply by AJenbo** [reply]
cold just take from the last loaded hero
